js/dialog.js

```diff
@@ -382,7 +387,15 @@
                   dataSource.responseType = YAHOO.util.DataSourceBase.TYPE_TEXT;
                   dataSource.parseTextData = function (request, response)
                     {
-                      return { 'results' : [ $(response).find('.label, #main-column h1').text().trim() ] };
+                      // For pages decorated with 'layout_2col' or 'layout_3col'
+                      var multiColumnResult = $(response).find('#main-column h1').text().trim();
+
+                      if (0 != multiColumnResult.length) {
+                        return { 'results' : [multiColumnResult] };
+                      } else {
+                        // Fallback for pages decorated with 'layout_1col'
+                        return { 'results' : [ $(response).find('h1').text().trim() ] };
```

Comment:
To avoid unexpected `h1` in some customized headers I'd use `#wrapper h1` as selector. I'd also use `.first()` before getting the text to make sure we have only the first one. And, a suggestion to avoid parsing the response twice into a jQuery var and some (really minor) duplication ...

```javascript
var $response = $(response);

// For pages decorated with 'layout_2col' or 'layout_3col'
var $multiColumnResult = $response.find('#main-column h1');

// Fallback for pages decorated with 'layout_1col'
if (0 == $multiColumnResult.length) {
  $multiColumnResult = $response.find('#wrapper h1');
}

return {'results': [$multiColumnResult.first().text().trim()]};
```

Reply:
I tried using the selector `#wrapper h1` prior to making the commit, but it doesn't seem to want to return anything. It still doesn't, and I have no idea why.

Alternatively, we could just add something like `id='resource-name'` to the `<h1>` we are targeting and use `#resource-name` as the selector.

Reply:
Right, I faced that issue a while ago and couldn't find the reason at first glance. Adding an id to the header sounds good @mInnes-archives, it may be needed later one in other pages, but we'll figure out. Thanks!